src/platform/nrfconnect/wifi/WiFiManager.cpp

```diff
@@ -145,302 +139,203 @@ CHIP_ERROR WiFiManager::Init()
         return CHIP_NO_ERROR;
     });
 
-    ChipLogDetail(DeviceLayer, "wpa_supplicant has been initialized");
+    net_mgmt_init_event_callback(&mWiFiMgmtClbk, WifiMgmtEventHandler, sWifiManagementEvents);
+    net_mgmt_add_event_callback(&mWiFiMgmtClbk);
+
+    ChipLogDetail(DeviceLayer, "WiFiManager has been initialized");
 
     return CHIP_NO_ERROR;
 }
-
-CHIP_ERROR WiFiManager::AddNetwork(const ByteSpan & ssid, const ByteSpan & credentials)
+CHIP_ERROR WiFiManager::Scan(const ByteSpan & ssid, ScanCallback callback)
 {
-    ChipLogDetail(DeviceLayer, "Adding WiFi network");
-    mpWpaNetwork = wpa_supplicant_add_network(wpa_s);
-    if (mpWpaNetwork)
-    {
-        static constexpr size_t kMaxSsidLen{ 32 };
-        mpWpaNetwork->ssid = (u8 *) k_malloc(kMaxSsidLen);
+    struct net_if * iface = net_if_get_default();
```

Comment:
Please use the GetInterface() helper function from Zephyr utils whenever accessing the net iface:
https://github.com/nrfconnect/sdk-connectedhomeip/blob/238a18db4eae15381c3c1838d8e863d841c59ead/src/platform/Zephyr/InetUtils.cpp#L32

Reply:
How can I obtain ifaceId there?

Reply:
You don't have to, the default argument is nullptr. Eventually it boils down to calling `net_if_get_default() `anyway, but we should have the unified way of obtaining the iface.